tool/change_metrics.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'dart:io';
+
+/// Calculates the amount of formatting changes in a given directory of code.
+///
+/// This should be run with a path to a directory. That directory should
+/// contain a Git repo. The committed state of the repo should be the formatted
+/// output before the change in question. Then the result of the new formatting
+/// should be unstaged changes.
+///
+/// Uses `git diff --shortstat` to calculate the number of changed lines.
+///
+/// Counts the number of lines of Dart code by reading the files.
+void main(List<String> arguments) {
+  if (arguments.length != 1) {
+    print('Usage: change_metrics.dart <dir>');
+    exit(1);
+  }
+
+  var directory = arguments[0];
+  var totalLines = 0;
+  var totalFiles = 0;
+
+  print('Counting lines...');
+  for (var entry in Directory(directory).listSync(recursive: true)) {
+    if (entry is File && entry.path.endsWith('.dart')) {
+      try {
+        var lines = entry.readAsLinesSync();
+        totalFiles++;
+        totalLines += lines.length;
+      } catch (error) {
+        print('Could not read ${entry.path}:\n$error');
+      }
+    }
+  }
+
+  print('Getting diff stats...');
+  var result = Process.runSync('git', ['diff', '--shortstat'],
+      workingDirectory: directory);
+  if (result.exitCode != 0) {
+    print('Git failure:\n${result.stdout}\n${result.stderr}');
+    exit(1);
+  }
+
+  var stdout = result.stdout as String;
+  var insertions = _parseGitStdout(stdout, r'(\d+) insertions');
+  var deletions = _parseGitStdout(stdout, r'(\d+) deletions');
+  var changes = insertions + deletions;
```

Comment:
Changes that are only on a single line are going to be double counted here - for example changing indentation in 1 line will show up as a delta of 2. This might also be fine to ignore as an approximation? I would expect that checking only inserted lines might be a better approximation for most changes though...

Reply:
I think it's OK if some lines are double counted. The goal is to give a rough measure of impact and over-estimating that is OK. It's probably generally better to be conservative with formatting changes.